## Assignment 1 Question 4
Use the file data.csv to implement multiple linear regression and understand the effect of
Investment on TV(x1), radio(x2) and news paper(x3) on sales(y). Using the model (multiple
linear regression) coefficients comment on null hypothesis.

In [3]:
import pandas as pd
import matplotlib.pyplot as plt
from sklearn import linear_model
import statsmodels.api as sm

In [4]:
df=pd.read_csv("data.csv",index_col="Unnamed: 0")
df.head()

TV  radio  newspaper  sales
1  230.1   37.8       69.2   22.1
2   44.5   39.3       45.1   10.4
3   17.2   45.9       69.3    9.3
4  151.5   41.3       58.5   18.5
5  180.8   10.8       58.4   12.9

In [9]:
X=df.drop("sales",axis=1)
y=df["sales"]

In [11]:
regr = linear_model.LinearRegression()
regr.fit(X, y)
print('Intercept: \n', regr.intercept_)
print('Coefficients: \n', regr.coef_)

Intercept: 
 2.938889369459412
Coefficients: 
 [ 0.04576465  0.18853002 -0.00103749]


In [13]:
#Add a constant to X
X = sm.add_constant(X)
 
model = sm.OLS(y, X).fit()
predictions = model.predict(X) 
 
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                  sales   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.896
Method:                 Least Squares   F-statistic:                     570.3
Date:                Sun, 20 Sep 2020   Prob (F-statistic):           1.58e-96
Time:                        21:47:00   Log-Likelihood:                -386.18
No. Observations:                 200   AIC:                             780.4
Df Residuals:                     196   BIC:                             793.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.9389      0.312      9.422      0.0

#### Conclusion
The ads on the newspaper, TV and Radio affect the sales as follows:
* Increasing the budget on only the TV ads increase the sales by 4%.
* Increasing the budget on only the radio increases the sales by 18%.
* Increasing the budget on only the newspaper does not affect the sales much.

#### Hypothesis testing
Null hypothesis says all the coefficient values are zero.
Alternate hypothesis says that atleast one coefficient value is non-zero.
The F static value is 570.3 which is greater than one which means it has strong evidence against the null hypothesis. The null hypothesis is rejected.